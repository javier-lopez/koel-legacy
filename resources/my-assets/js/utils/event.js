import event from '@phanan/vuebus'
import { events as eventNames } from '@/config'

event.$names = eventNames

export { event }
