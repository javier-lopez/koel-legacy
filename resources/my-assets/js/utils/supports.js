import isMobile from 'ismobilejs'

/**
 * Check if AudioContext is supported by the current browser.
 *
 * @return {Boolean}
 */
export const isAudioContextSupported = (() => {
  // Apple device just doesn't love AudioContext that much.
  if (isMobile.apple.device) {
    return false
  }

  const AudioContext = (window.AudioContext ||
    window.webkitAudioContext ||
    window.mozAudioContext ||
    window.oAudioContext ||
    window.msAudioContext)

  if (!AudioContext) {
    return false
  }

  // Safari (MacOS & iOS alike) has webkitAudioContext, but is buggy.
  // @link http://caniuse.com/#search=audiocontext
  if (!(new AudioContext()).createMediaElementSource) {
    return false
  }

  return true
})()

/**
 * Checks if HTML5 clipboard can be used.
 * @return {Boolean}
 */
export const isClipboardSupported = 'execCommand' in document

/**
 * Checks if Media Session API is supported.
 * @return {Boolean}
 */
export const isMediaSessionSupported = 'mediaSession' in navigator
