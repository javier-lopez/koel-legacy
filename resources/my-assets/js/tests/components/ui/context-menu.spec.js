import Component from '@/components/ui/context-menu'
import { event } from '@/utils'
import { mock } from '@/tests/__helpers__'

describe('components/ui/context-menu', () => {
  afterEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders', () => {
    expect(mount(Component)).toMatchSnapshot()
  })

  it('renders extra CSS classes', () => {
    const wrapper = mount(Component, {
      propsData: {
        extraClass: 'foo'
      }
    })
    expect(wrapper.find('.menu').hasClass('foo')).toBe(true)
  })

  it('opens', () => {
    const wrapper = mount(Component)
    wrapper.vm.open(42, 128)
    expect(wrapper.find('.menu').element.style.top).toBe('42px')
    expect(wrapper.find('.menu').element.style.left).toBe('128px')
    expect(getComputedStyle(wrapper.find('.menu').element).display).toBe('block')
  })

  it('closes', () => {
    const wrapper = mount(Component)
    wrapper.vm.open(42, 128)
    wrapper.vm.close()
    expect(getComputedStyle(wrapper.find('.menu').element).display).toBe('none')
  })

  it('notifies other instances to close', () => {
    const m = mock(event, 'emit')
    const wrapper = mount(Component)
    wrapper.vm.open(42, 128)
    expect(m).toHaveBeenCalledWith('CONTEXT_MENU_OPENING')
  })
})
