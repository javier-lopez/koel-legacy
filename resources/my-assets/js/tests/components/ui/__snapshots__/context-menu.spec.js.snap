// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/context-menu renders 1`] = `<ul tabindex="-1" class="menu" style="top: 0px; left: 0px; display: none;">Menu items to go here.</ul>`;
